src/System.Private.ServiceModel/tools/test/Bridge/WcfTestBridgeCommon/CertificateManager.cs

```diff
@@ -0,0 +1,236 @@
+
+            string path = Path.Combine(Path.Combine(configuration.BridgeResourceFolder, "Certificates"), certificateName);
+            if (!File.Exists(path))
+            {
+                throw new InvalidOperationException(String.Format("The requested certificate file at {0} does not exist.",
+                                                    path));
+            }
+
+            return path;
+        }
+
+        // Install the certificate in the given file path into the Root store and returns its thumbprint
+        public static string InstallRootCertificate(BridgeConfiguration configuration, string certificateName)
+        {
+            string certificateFilePath = CreateCertificateFilePath(configuration, certificateName);
+
+            lock (s_certificateLock)
+            {
+                foreach (var pair in s_rootCertificates)
+                {
+                    if (string.Equals(certificateName, pair.Value, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return pair.Key;
+                    }
+                }
+
+                X509Store store = new X509Store(StoreName.Root, StoreLocation.LocalMachine);
+                store.Open(OpenFlags.ReadWrite);
+                X509Certificate2 cert = new X509Certificate2(certificateFilePath);
+                store.Add(cert);
+                store.Close();
+
+                s_rootCertificates[cert.Thumbprint] = certificateName;
+                return cert.Thumbprint;
+            }
+        }
+
+        // Install the certificate in the given file path into the My store
+        public static string InstallMyCertificate(BridgeConfiguration configuration, string certificateName)
+        {
+            // Installing any certificate guarantees the certificate authority is loaded first
+            InstallRootCertificate(configuration, configuration.BridgeCertificateAuthority);
+
+            string certificateFilePath = CreateCertificateFilePath(configuration, certificateName);
+
+            lock (s_certificateLock)
+            {
+                foreach (var pair in s_myCertificates)
+                {
+                    if (string.Equals(certificateName, pair.Value, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return pair.Key;
+                    }
+                }
+
+                X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine);
+                store.Open(OpenFlags.ReadWrite);
+
+                X509Certificate2 cert = new X509Certificate2();
+                cert.Import(certificateFilePath, String.Empty, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet);
+                store.Add(cert);
+                store.Close();
+
+                s_myCertificates[cert.Thumbprint] = certificateName;
+                return cert.Thumbprint;
+            }
+        }
+
+        private static void UninstallAllRootCertificates()
+        {
```

Comment:
Although not a blocker for this PR, I would like to see it in a follow up one. We should have some mechanism to identify left behind certificates from a previous run to be able to clean them up. Similar to how the firewall rules are done.